src/coinbase/address/wallet_address.ts

```diff
@@ -340,4 +352,251 @@ export class WalletAddress extends Address {
+   */
+  private async createStakingOperation(
+    amount: Amount,
+    assetId: string,
+    action: string,
+    mode: StakeOptionsMode,
+    options: { [key: string]: string },
+    timeoutSeconds: number,
+    intervalSeconds: number,
+  ): Promise<StakingOperation> {
+    let stakingOperation = await this.createStakingOperationRequest(
+      amount,
+      assetId,
+      action,
+      mode,
+      options,
+    );
+
+    // NOTE: Staking does not yet support server signers at this point.
+    await stakingOperation.sign(this.key!);
+    for (let i = 0; i < stakingOperation.getTransactions().length; i++) {
+      const transaction = stakingOperation.getTransactions()[0];
+      if (!transaction.isSigned()) {
+        continue;
+      }
+      stakingOperation = await this.broadcastStakingOperationRequest(
+        stakingOperation,
+        transaction.getSignedPayload()!.slice(2),
+        i,
+      );
+    }
+
+    const startTime = Date.now();
+    while (Date.now() - startTime < timeoutSeconds * 1000) {
+      await stakingOperation.reload();
+      const status = stakingOperation.getStatus();
+      if (
+        status === StakingOperationStatusEnum.Complete ||
+        status === StakingOperationStatusEnum.Failed
+      ) {
+        return stakingOperation;
+      }
+      await delay(intervalSeconds);
+    }
+    throw new Error("Staking Operation timed out");
+  }
+
+  /**
+   * Lists the staking rewards for the address.
+   *
+   * @param assetId - The asset ID.
+   * @param startTime - The start time.
+   * @param endTime - The end time.
+   * @param format - The format to return the rewards in. (usd, native). Defaults to usd.
+   * @returns The staking rewards.
+   */
+  public async stakingRewards(
+    assetId: string,
+    startTime = getWeekBackDate(new Date()),
+    endTime = formatDate(new Date()),
+    format: StakingRewardFormat = StakingRewardFormat.Usd,
+  ): Promise<StakingReward[]> {
+    return StakingReward.list(
+      Coinbase.normalizeNetwork(this.getNetworkId()),
+      assetId,
+      [this.getId()],
+      startTime,
+      endTime,
+      format,
+    );
+  }
+
+  /**
+   * A helper function that creates the staking operation.
+   *
+   * @param amount - The amount for the staking operation.
+   * @param assetId - The asset for the staking operation.
+   * @param action - The type of staking action to perform.
+   * @param mode - The staking mode. Defaults to DEFAULT.
+   * @param options - Additional options such as setting the mode for the staking action.
+   * @private
+   * @throws {Error} if the amount is less than zero.
+   * @returns The created staking operation.
+   */
+  private async createStakingOperationRequest(
+    amount: Amount,
+    assetId: string,
+    action: string,
+    mode: StakeOptionsMode = StakeOptionsMode.DEFAULT,
+    options: { [key: string]: string } = {},
+  ): Promise<StakingOperation> {
+    if (new Decimal(amount.toString()).lessThanOrEqualTo(0)) {
```

Comment:
this feels like an unnecessary hard check deep in the stack..
can the amount specific checks happen higher up like in the `validateCanClaimStake` etc funcc ?